weather_forecast_retrieval/hrrr_preprocessor.py

```diff
@@ -63,6 +74,29 @@ def __init__(self, hrrr_dir, start_date, end_date, output_dir,
         self._logger.info('Forecast hour: {}'.format(self.forecast_hr))
         self._logger.info('Number of cpu argument: {}'.format(self.ncpu))
 
+    @property
+    def variable_match(self):
+        return '|'.join(self.VARIABLES)
+
+    def check_for_good_file(self, file_name):
+
+        check_action = 'wgrib2 {}'.format(file_name)
+        status, output = self.call_wgrib2(check_action)
+
+        bad_flag = False
+        if status != 0:
+            bad_flag = True
+        else:
+            output = ''.join(output)
+            for variable in self.VARIABLES:
+                if variable not in output:
+                    self._logger.warning('Variable {} not in file'.format(variable))
+                    bad_flag = True
```

Comment:
Minor, but you can stop the iteration once you detect one bad flag.

Reply:
Agreed, I like seeing what was bad in the log but probably doesn't matter to much.

Reply:
If verbose is on then it will print all else it will break the loop on the first warning.